packages/datadog-plugin-google-cloud-pubsub/src/pubsub-transit-handler.js

```diff
@@ -120,13 +120,66 @@ class GoogleCloudPubsubTransitHandlerPlugin extends TracingPlugin {
         const producerParent = this.tracer.extract('text_map', attrs) || null
         const effectiveParent = producerParent || undefined
 
-        // ToDo: create pubsub.delivery; create HTTP span directly as child of producer
+        // Compute pubsub scheduling duration (publish → HTTP receipt)
+        const publishStartTimeRawForHttp = attrs['x-dd-publish-start-time']
+        let schedulingMs = null
+        if (publishStartTimeRawForHttp) {
+          const t0 = Number.parseInt(publishStartTimeRawForHttp, 10)
+          if (Number.isFinite(t0) && t0 > 0) schedulingMs = Date.now() - t0
+        }
+
+        // Create pubsub.delivery span to represent infra delivery latency and parent the HTTP span
+        let parentForHttp = effectiveParent
+        const publishStartTimeRaw = attrs['x-dd-publish-start-time']
+        if (publishStartTimeRaw) {
+          const publishStartTime = Number.parseInt(publishStartTimeRaw, 10)
+          if (Number.isFinite(publishStartTime) && publishStartTime > 0) {
+            const messageId = (message && message.messageId) || req.headers['ce-id']
+            const deliveryTags = {
```

Comment:
```suggestion
            const tags = {
```

I think it's fine to use the shorter name. We always want to deliver tags :)